orderBook.js

```diff
@@ -1,5 +1,45 @@
-function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+/* eslint-disable indent */
+const reconcileOrder = (existingBook, incomingOrder) => {
+
+    if (existingBook.length === 0) {
+        existingBook.push(incomingOrder)
+
+    } else {
+
+        const matchingOrders = existingBook.find(order => order.type !== incomingOrder.type && order.price === incomingOrder.price)
+        const matchingOrdersIndex = existingBook.findIndex(order => order.type !== incomingOrder.type && order.price === incomingOrder.price)
+        const partialMatching = existingBook.find(order => order.type !== incomingOrder.type && order.price > incomingOrder.price && order.quantity === incomingOrder.quantity)
+        const partialMatchingIndex = existingBook.findIndex(order => order.type !== incomingOrder.type && order.price > incomingOrder.price && order.quantity === incomingOrder.quantity)
```

Comment:
Nice use of find and findIndex. One thing that could clean this up is to only use findIndex and then use the index to get the corresponding order.

```JavaScript
const matchingOrdersIndex = existingBook.findIndex(order => order.type !== incomingOrder.type && order.price === incomingOrder.price)
const matchingOrder = existingBook[matchingOrdersIndex]
```